ext/dom/php_dom.c

```diff
@@ -1570,14 +1569,14 @@ void dom_nnodemap_objects_free_storage(zend_object *object) /* {{{ */
 		if (objmap->cached_obj && GC_DELREF(&objmap->cached_obj->std) == 0) {
 			zend_objects_store_del(&objmap->cached_obj->std);
 		}
-		if (objmap->free_local) {
-			xmlFree(objmap->local);
+		if (objmap->release_local) {
+			zend_string_release((zend_string *) (objmap->local - XtOffsetOf(zend_string, val)));
```

Comment:
Maybe add a helper for this, don't remember for what other extension we did this.

Reply:
Yeah, I'll add the one from ldao